stl/inc/format

```diff
@@ -259,12 +269,16 @@ public:
 
     public:
         template <class _Ty>
-        explicit handle(const _Ty& _Val) noexcept
+        explicit handle(_Ty&& _Val) noexcept
             : _Ptr(_STD addressof(_Val)),
               _Format([](basic_format_parse_context<_CharType>& _Parse_ctx, _Context& _Format_ctx, const void* _Ptr) {
-                  typename _Context::template formatter_type<_Ty> _Formatter;
+                  using _Value_type = remove_cvref_t<_Ty>;
+                  typename _Context::template formatter_type<_Value_type> _Formatter;
+                  using _Qualified_type =
+                      conditional_t<_Has_const_formatter<_Value_type, _Context>, const _Value_type, _Value_type>;
                   _Parse_ctx.advance_to(_Formatter.parse(_Parse_ctx));
-                  _Format_ctx.advance_to(_Formatter.format(*static_cast<const _Ty*>(_Ptr), _Format_ctx));
+                  _Format_ctx.advance_to(_Formatter.format(
+                      *const_cast<_Qualified_type*>(static_cast<const _Value_type*>(_Ptr)), _Format_ctx));

```

Comment:
I think this is covered by the requirement that remove_cvref_t<T> be formattable to be put in a handle (which is not what this code does, because said wording was a follow up issue (LWG3631).

If remove_cvref_t<T> still has const qualifiers then const-formattable is the same as formattable (again, that's my understanding)

Reply:
I wouldn't mind some input from Casey here

Reply:
Resolving as this was a real issue (yay! :smile_cat:) that has been fixed by a commit.